### PR TITLE
add ASP.NET know issues for .NET 4.8 release

### PR DESCRIPTION
**Pull request by HongGit**

### CODE REVIEW

File: Documentation/KnownIssues/776516-InvalidOperationException in System_Web_ni!System.Web.Hosting.RecycleLimitMonitor+RecycleLimitMonitorSingleton.AlertProxyMonitors.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# InvalidOperationException in ASP.Net-RecycleLimitMonitor-AlertProxyMonitors
```
Comment (rpetrusha) [inline Documentation/KnownIssues/776516-InvalidOperationException in System_Web_ni!System.Web.Hosting.RecycleLimitMonitor+RecycleLimitMonitorSingleton.AlertProxyMonitors.md]:
```suggestion
# InvalidOperationException in ASP.NET RecycleLimitMonitor-AlertProxyMonitors
```

File: Documentation/KnownIssues/776516-InvalidOperationException in System_Web_ni!System.Web.Hosting.RecycleLimitMonitor+RecycleLimitMonitorSingleton.AlertProxyMonitors.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+## Symptoms
+ASP.Net applications may crash intermittently due to an `InvalidOperationException` in
```
Comment (rpetrusha) [inline Documentation/KnownIssues/776516-InvalidOperationException in System_Web_ni!System.Web.Hosting.RecycleLimitMonitor+RecycleLimitMonitorSingleton.AlertProxyMonitors.md]:
```suggestion
ASP.NET applications may crash intermittently due to an `InvalidOperationException` in
```

File: Documentation/KnownIssues/776516-InvalidOperationException in System_Web_ni!System.Web.Hosting.RecycleLimitMonitor+RecycleLimitMonitorSingleton.AlertProxyMonitors.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+`System_Web_ni!System.Web.Hosting.RecycleLimitMonitor+RecycleLimitMonitorSingleton.AlertProxyMonitors()`.
+
+## Cause
+This issue was introduced when ASP.Net 4.7 separated the concerns of cache management and memory monitoring. This new
```
Comment (rpetrusha) [inline Documentation/KnownIssues/776516-InvalidOperationException in System_Web_ni!System.Web.Hosting.RecycleLimitMonitor+RecycleLimitMonitorSingleton.AlertProxyMonitors.md]:
```suggestion
This issue was introduced when ASP.Net 4.7 separated cache management and memory monitoring. This new
```

File: Documentation/KnownIssues/776516-InvalidOperationException in System_Web_ni!System.Web.Hosting.RecycleLimitMonitor+RecycleLimitMonitorSingleton.AlertProxyMonitors.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
+# InvalidOperationException in ASP.Net-RecycleLimitMonitor-AlertProxyMonitors
+
+## Symptoms
+ASP.Net applications may crash intermittently due to an `InvalidOperationException` in
+`System_Web_ni!System.Web.Hosting.RecycleLimitMonitor+RecycleLimitMonitorSingleton.AlertProxyMonitors()`.
```
Comment (rpetrusha) [inline Documentation/KnownIssues/776516-InvalidOperationException in System_Web_ni!System.Web.Hosting.RecycleLimitMonitor+RecycleLimitMonitorSingleton.AlertProxyMonitors.md]:
```suggestion
`System_Web_ni!System.Web.Hosting.RecycleLimitMonitor+RecycleLimitMonitorSingleton.AlertProxyMonitors`.
```

File: Documentation/KnownIssues/776516-InvalidOperationException in System_Web_ni!System.Web.Hosting.RecycleLimitMonitor+RecycleLimitMonitorSingleton.AlertProxyMonitors.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+## Cause
+This issue was introduced when ASP.Net 4.7 separated the concerns of cache management and memory monitoring. This new
+`RecycleLimitMonitorSingleton` monitors worker process private bytes usage against the recycling limit for the process, and when
+the limit is near, it alerts a set of "proxy monitors" in each registered AppDomain so they can take action to avoid a process
+recycle. The collection of "proxy monitors" being enumerated was not thread-safe though, so new AppDomains spinning up or old
```
Comment (rpetrusha) [inline Documentation/KnownIssues/776516-InvalidOperationException in System_Web_ni!System.Web.Hosting.RecycleLimitMonitor+RecycleLimitMonitorSingleton.AlertProxyMonitors.md]:
```suggestion
recycle. The collection of "proxy monitors" being enumerated was not thread-safe, though, so new AppDomains spinning up or old
```

File: Documentation/KnownIssues/776516-InvalidOperationException in System_Web_ni!System.Web.Hosting.RecycleLimitMonitor+RecycleLimitMonitorSingleton.AlertProxyMonitors.md [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+AppDomains shutting down while this monitor enumerated the collection could result in this unhandled `InvalidOperationException`
+in the default AppDomain, causing a crash.
+
+## Workaround 
+There is no reliable workaround. If the environment does not tend to consume private bytes near the recycling limit, or if
+web app domain lifetimes tend to last the entire length of the process lifetime, then chances of hitting this issue are
+extremely minimal. More private bytes consumption and/or more AppDomain recycling will increase the odds of hitting this.
```
Comment (rpetrusha) [inline Documentation/KnownIssues/776516-InvalidOperationException in System_Web_ni!System.Web.Hosting.RecycleLimitMonitor+RecycleLimitMonitorSingleton.AlertProxyMonitors.md]:
```suggestion
extremely minimal. More private bytes consumption and/or more AppDomain recycling increases the odds of hitting this exception condition.
```

File: Documentation/KnownIssues/776516-InvalidOperationException in System_Web_ni!System.Web.Hosting.RecycleLimitMonitor+RecycleLimitMonitorSingleton.AlertProxyMonitors.md [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+
+## Resolution
+A hotfix for this issue is in the works for .Net 4.8. This page will be updated with a link to the hotfix when it becomes
```
Comment (rpetrusha) [inline Documentation/KnownIssues/776516-InvalidOperationException in System_Web_ni!System.Web.Hosting.RecycleLimitMonitor+RecycleLimitMonitorSingleton.AlertProxyMonitors.md]:
```suggestion
A hotfix for this issue is planned for .Net 4.8. This page will be updated with a link to the hotfix when it becomes
```

**Comment by rpetrusha** [comment]
@HongGit, this PR also targets master rather than net48. Could you do the following:
1. Make your changes to the file, then save the new file somewhere outside of GitHub.
2. Close this PR.
3. Create a new branch based on the net48 branch rather than master.
4. Add the new file to it.
5. Open a new PR.

File: Documentation/KnownIssues/776516-InvalidOperationException in System_Web_ni!System.Web.Hosting.RecycleLimitMonitor+RecycleLimitMonitorSingleton.AlertProxyMonitors.md (hunk 7 above)
Comment (HongGit) [reply Documentation/KnownIssues/776516-InvalidOperationException in System_Web_ni!System.Web.Hosting.RecycleLimitMonitor+RecycleLimitMonitorSingleton.AlertProxyMonitors.md]: @rpetrusha can you please give me the link to the net48 branch?  

I was confused when I created this PR, this is the link from the email I received.

Comment (rpetrusha) [reply Documentation/KnownIssues/776516-InvalidOperationException in System_Web_ni!System.Web.Hosting.RecycleLimitMonitor+RecycleLimitMonitorSingleton.AlertProxyMonitors.md]: If you're working with bash, you have to check out the **net48** branch, then create a new branch that based on it. When you open the PR in GitHub, target the **net48** branch, as shown in this image:
image
Let me know if that doesn't help or you're still having difficulty, @HongGit.

**Comment by HongGit** [comment]
per @rpetrusha close this PR.

**Pull request closed without merging**